File: isegoria/src/main/java/com/eulersbridge/isegoria/auth/signup/SimpleOnItemSelectedListener.java
package com.eulersbridge.isegoria.auth.signup;

import android.view.View;
import android.widget.AdapterView;

class SimpleOnItemSelectedListener implements AdapterView.OnItemSelectedListener {

    void onItemSelected(int position) {
        throw new RuntimeException("Stub! Override this method.");
    }

    @Override
    public void onItemSelected(AdapterView<?> adapterView, View view, int position, long id) {
        onItemSelected(position);
    }

    @Override
    public void onNothingSelected(AdapterView<?> adapterView) {
        // Ignored, unused
    }
}

File: isegoria/src/main/java/com/eulersbridge/isegoria/auth/signup/SimpleOnItemSelectedListener.kt
package com.eulersbridge.isegoria.auth.signup

import android.view.View
import android.widget.AdapterView

internal open class SimpleOnItemSelectedListener : AdapterView.OnItemSelectedListener {

    internal open fun onItemSelected(position: Int) {
        throw RuntimeException("Stub! Override this method.")
    }

    override fun onItemSelected(adapterView: AdapterView<*>, view: View, position: Int, id: Long) {
        onItemSelected(position)
    }

    override fun onNothingSelected(adapterView: AdapterView<*>) {
        // Ignored, unused
    }
}
